MyAwesomeProject/components/RegisterUser.tsx

```diff
@@ -0,0 +1,223 @@
+              autoCapitalize = 'none'
+              onChangeText={this.handleEmailChange}/>
+            <Text style={styles.errorText}>{this.state.emailError}</Text>
+ 
+            <Text style={styles.formsText}>CPF</Text>
+            <TextInput style={styles.inputBox} 
+              autoCapitalize = 'none'
+              onChangeText={this.handleCpfChange}
+              placeholder='Somente Números'/>
+            <Text style={styles.errorText}>{this.state.cpfError}</Text>
+
+            <Text style={styles.formsText}>Data de Nascimento</Text>
+            <TextInput style={styles.inputBox}
+              autoCapitalize = 'none'
+              onChangeText={this.handleBirthDateChange}
+              placeholder='Formato YYYY-MM-DD'/>
+            <Text style={styles.errorText}>{this.state.birthDateError}</Text>
+
+            <Text style={styles.formsText}>Senha</Text>
+            <TextInput style={styles.inputBox}
+              autoCapitalize = 'none'
+              onChangeText={this.handlePasswordChange}/>
+            <Text style={styles.errorText}>{this.state.passwordError}</Text>
+            
+            <Text style={styles.formsText}>Função</Text>
+            <Picker
+              selectedValue={this.state.role}
+              style={styles.picker}
+              itemStyle={styles.itemStyle}
+              onValueChange={this.handleRoleChange}>
+              <Picker.Item label="Usuário" value="user"/>
+              <Picker.Item label="Administrador" value="admin"/>
+            </Picker>
+
+            <View style={styles.button}>
+              <Button
+                title="Cadastrar"
+                onPress={this.handleButtonTap}
+                disabled={this.state.isLoading}
+                color="white"
+              />
+            </View>
+
+            <View style={styles.loading}>
+              {this.state.isLoading && <ActivityIndicator size="large" color="black"/>}            
+            </View>
+        </View>
+      </View>
+    )
+  }
+
+  private handleNameChange = (text: string) => this.name = text
+  private handleEmailChange = (text: string) => this.email = text
+  private handlePasswordChange = (text: string) => this.password = text
+  private handleCpfChange = (text: string) => this.cpf = text
+  private handleBirthDateChange = (text: string) => this.birthDate = text
+  private handleRoleChange = (text: string) => this.setState({role: text})
+
+  private handleButtonTap = () => {
+    const nameErrorText: string = validateName(this.name)
+    const emailErrorText: string = validateEmail(this.email)
+    const passwordErrorText: string = validatePassword(this.password)
+    const cpfErrorText: string = validateCpf(this.cpf)
+    const birthDateErrorText: string = validateBirthDate(this.birthDate)
+
+    if (!nameErrorText && !emailErrorText && !passwordErrorText && !cpfErrorText && !birthDateErrorText) {
+      this.setState({isLoading: true})
+      this.doRegisterUser()
```

Comment:
Como o seu `doRegisterUser` faz uma chamada, é razoável pensar que sempre que ela for feita o estado de load deveria ser setado, concorda? Não acha melhor então mandar a linha de cima pra dentro desse método?

Reply:
Tem razão